x-pack/plugins/security_solution/public/management/pages/policy/view/policy_list.tsx

```diff
@@ -89,7 +89,9 @@ export const PolicyList = memo(() => {
   const policyIdToEndpointCount = useMemo(() => {
     const map = new Map<AgentPolicy['package_policies'][number], number>();
     for (const policy of endpointCount?.items) {
-      map.set(policy.package_policies[0], policy.agents ?? 0);
+      for (const packagePolicy of policy.package_policies) {
+        map.set(packagePolicy, policy.agents ?? 0);
```

Comment:
Maybe its just me, but I found the variable names very confusing here. `endpointCount` not actually being endpoint counts... `policy` (I think) really being fleetAgentPolicy. Also, I'm wondering if this logic here should really be moved to the hook `useGetAgentCountForPolicy()` - based on that hook's name, it sounds like it should be responsible for actually returning counts for each policy id it was provided.

I understand this PR is fixing a bug only and need to keep the scope low since its being back ported, but consider this for a follow up into `main` as tech debt.

______

Suggestion: only store package policy that we care about and not every other package policy in the agent policy. I think this would work:

```typescript
if (policyIds.includes(packagePolicy) {
  map.set(packagePolicy, policy.agents ?? 0);
}
```

_(`packagePolicy` is the ID only, right? or the entire package policy record? if the entire record, then just use `packagePolicy.id` in the `if()` statement)_